components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/openidconnect/DefaultOIDCClaimsCallbackHandler.java

```diff
@@ -187,10 +187,12 @@ public void handleCustomClaims(JWTClaimsSet jwtClaimsSet, OAuthAuthzReqMessageCo
         // TODO: Get claims filtered by essential claims and add to returning claims
         // https://github.com/wso2/product-is/issues/2680
 
-        // Handle essential claims of the request object
-        Map<String, Object> claimsFromRequestObject =
-                filterClaimsFromRequestObject(userClaimsInOIDCDialect, accessToken);
-        filteredUserClaimsByOIDCScopes.putAll(claimsFromRequestObject);
+        if (accessToken != null) {
```

Comment:
Can we have  a log here if the accessToken was null to say that we skipped filtering claims based on the request object?

Reply:
Added

Reply:
As you have mentioned in the offline I am skipping this since this log is not suitable for all the flows.